app/scripts/wizard.js

```diff
@@ -70,12 +71,16 @@ function WizardDirective() {
                 }
 
                 $scope.goTo = function (step) {
-                    unselectAll();
-                    $scope.selectedStep = step;
-                    if (!_.isUndefined($scope.currentStep)) {
-                        $scope.currentStep = step.code;
+                    $scope.nextStep = step.code;
+
```

Comment:
Why do you need `$scope.nextStep`, `this.goToNextStep`, `$scope.canChangeStep`. Doesn't it work to call `$scope.canTabChange` from `this.goTo` directly?

Reply:
`this.goTo` is called when we want to go to a certain step without clicking on the tab, otherwise `$scope.goTo` is called directly on clicking a tab. `$scope.nextStep` holds the next steps name, so when the user closes the confirmation prompt and wants to change the step, then `$scope.goToNextStep` method takes him to the step he clicked earlier.

Reply:
This is how `canTabChange` method looks
![code_cut](https://user-images.githubusercontent.com/10600741/61292212-3e40cd00-a7ee-11e9-8625-a7e697416009.png)
 In this method, if in place of `$q.when(true)`, I try to use `Promise.resolve(true)` then I had to double click on the same tab to get it changed. I don't understand why it's happening. Do you have any idea?

Reply:
Firstly, I would avoid using angular $q promises together with native promises. It might not be the issue, but it's a good practice not to mix things. Prefer native promises where possible.

It's also strange that when `confirmSave` is resolved, you are resolving `false` from `canTabChange`. If you still can't figure it out, push the non-working code and I'll take a look.

Reply:
Sorry, my bad, that `Promise.resolve(confirmSave)` is not there, by mistake I left it there. The issue is with `Promise.resolve(true)` instead of `$q.when(true)`.

Reply:
I don't see any other obvious mistakes. If you can't figure it out, post the commit tags of ui-framework and client-core and I'll check it out.

Reply:
client-core: `eb6a559`
ui-framework: `51b1a53`

Reply:
I don't see a PR in client-core. Can you push the code so I can check it out?

Reply:
sorry, this is the PR https://github.com/superdesk/superdesk-client-core/pull/3000

Reply:
![screenshot_1](https://user-images.githubusercontent.com/2076953/61354240-067b6900-a872-11e9-9461-2c89b4190405.png)

I found the issue. You need to add `$scope.$apply();` to tell angular to re-render. You don't need that when using angular $q promises, because they are coupled with angular and always tell it to re-render when promises resolve.